lldb/source/Interpreter/ScriptInterpreter.cpp

```diff
@@ -106,10 +106,13 @@ ScriptInterpreter::GetOpaqueTypeFromSBEvent(const lldb::SBEvent &event) const {
   return event.m_opaque_ptr;
 }
 
-Stream *ScriptInterpreter::GetOpaqueTypeFromSBStream(
+lldb::StreamSP ScriptInterpreter::GetOpaqueTypeFromSBStream(
     const lldb::SBStream &stream) const {
-  if (stream.m_opaque_up)
-    return const_cast<lldb::SBStream &>(stream).m_opaque_up.get();
+  if (stream.m_opaque_up) {
+    lldb::StreamSP s = std::make_shared<lldb_private::StreamString>();
+    *s << const_cast<lldb::SBStream &>(stream).GetData();
```

Comment:
This adds a dependency cycle: lldb::SBStream is in source/API, so this makes Interpreter depend on API. But API already depends on Interpreter.

Reply:
This isn't an idle complaint: It has the effect that `lldb-test` no longer links after this PR if you don't pass `-Wl,-dead_strip` to the linker.

Reply:
Chances are it also breaks building on Windows, where /OPT:REF might not be strong enough to remove this.

Also, we do have a `LLVM_NO_DEAD_STRIP` option, which presumably no longer builds after this change either.

Reply:
Reverted in a2e3af5d581547d3ea53e5383d6f7f1cab45120a for now.

Reply:
Looks like it broke the windows bot: https://lab.llvm.org/buildbot/#/builders/141/builds/369

@nico Thanks to the revert, I think I have a fix for this.